compiler/src/dotty/tools/dotc/printing/Texts.scala

```diff
@@ -155,7 +171,7 @@ object Texts {
     def lines(xs: Traversable[Text]) = Vertical(xs.toList.reverse)
   }
 
-  case class Str(s: String) extends Text {
+  case class Str(s: String, startLine: Int = Int.MaxValue, endLine: Int = -1) extends Text {
```

Comment:
I would create a special purpose `class Lines` with a special instance `NoLine`. I think it is weird to use `Int.MaxValue` and `-1` as special values. You can even make it a value class that wraps a `Long` like `Position` if performance matters here.